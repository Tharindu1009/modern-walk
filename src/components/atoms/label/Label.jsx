import './styles/label.scss';
import * as React from 'react';
import CN from "classnames";
import { labelTypes } from '../../../constants';

function Label({ type, text }) {

    // Appling relevant style class by checking the "type" prop
    const LabelClasses = CN({
        "label label__header": type === labelTypes.header,
        "label label__sectionHeader": type === labelTypes.sectionHeader,
        "label label__cardHeader": type === labelTypes.cardHeader,
        "label label__productPrice": type === labelTypes.productPrice,
        "label label__productDesc": type === labelTypes.productDesc,
    });

    return (
        <div data-testid="label" className={LabelClasses}>
            {type ===  labelTypes.productPrice && "Rs "} {text}
        </div>
    );
}

export default Label;

