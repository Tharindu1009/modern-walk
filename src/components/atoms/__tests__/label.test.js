import { render, cleanup } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import Label from "../label/Label";

describe('Label', () => {
    afterEach(() => {
        cleanup();
    })

    it('should render header label component correctly', () => {
        const { getByTestId } = render(<Label type="header" text="Modern Walk" />);
        const labelElement = getByTestId('label');
        expect(labelElement).toBeInTheDocument();
    });

    it('should render section header label component correctly with Flash Sale text', () => {
        const { getByTestId } = render(<Label type="sectionHeader" text="Flash Sale" />);
        const labelElement = getByTestId('label');
        expect(labelElement).toBeInTheDocument();
    });

    it('should render section header label component correctly with Categories text', () => {
        const { getByTestId } = render(<Label type="sectionHeader" text="Categories" />);
        const labelElement = getByTestId('label');
        expect(labelElement).toBeInTheDocument();
    });

    it('should render category banner label component correctly with Mens Clothing text', () => {
        const { getByTestId } = render(<Label type="categoryBanner" text="Men's Clothing" />);
        const labelElement = getByTestId('label');
        expect(labelElement).toBeInTheDocument();
    });

    it('should render category banner label component correctly with Womens Clothing text', () => {
        const { getByTestId } = render(<Label type="categoryBanner" text="Women's Clothing" />);
        const labelElement = getByTestId('label');
        expect(labelElement).toBeInTheDocument();
    });

    it('matches label snapshot', () => {
        const tree = TestRenderer.create(<Label type="header" text="Modern Walk" />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});