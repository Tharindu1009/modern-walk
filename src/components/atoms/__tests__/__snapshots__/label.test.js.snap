// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label matches label snapshot 1`] = `
<div
  className="label label__header"
  data-testid="label"
>
   
  Modern Walk
</div>
`;
